packages/replay-next/src/suspense/HitPointsCache.ts

```diff
@@ -0,0 +1,138 @@
+import { Location, PointRange, TimeStampedPoint } from "@replayio/protocol";
+
+import { MAX_POINTS_FOR_FULL_ANALYSIS } from "protocol/thread/analysis";
+import { createFetchAsyncFromFetchSuspense } from "replay-next/src/utils/suspense";
+import {
+  HitPointStatus,
+  HitPointsAndStatusTuple,
+  ReplayClientInterface,
+} from "shared/client/types";
+import { isThennable } from "shared/proxy/utils";
+import { ProtocolError, isCommandError } from "shared/utils/error";
+
+import { RangeCache, createGenericRangeCache } from "./createGenericRangeCache";
+
+const hitPointCaches = new Map<string, RangeCache<TimeStampedPoint>>();
+
+function getHitPointsCache(location: Location, condition: string | null) {
+  const key = `${location.sourceId}:${location.line}:${location.column}:${condition}`;
+  if (!hitPointCaches.has(key)) {
+    hitPointCaches.set(key, createHitPointsCache(location, condition));
+  }
+  return hitPointCaches.get(key)!;
+}
+
+function createHitPointsCache(location: Location, condition: string | null) {
+  return createGenericRangeCache<TimeStampedPoint>(
+    async (client, range, cacheValues, cacheError) => {
+      const locations = client.getCorrespondingLocations(location).map(location => ({
+        location,
+      }));
+
+      if (condition) {
+        const mapper = `
```

Comment:
Shouldn't this be typed code now, after @markerikson's recent change?

Reply:
This mapper hadn't been converted yet.